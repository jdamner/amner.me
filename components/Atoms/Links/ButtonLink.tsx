import React from "react";

/* Components */
import Link, { type LinkProps } from "next/link";
import { ButtonClasses } from "./Button";

export default function ButtonLink(
  props: Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, keyof LinkProps> &
    LinkProps & {
      children?: React.ReactNode;
    } & React.RefAttributes<HTMLAnchorElement>,
) {
  return <Link className={ButtonClasses.join(" ")} {...props} />;
}
