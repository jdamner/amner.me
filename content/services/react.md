---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
