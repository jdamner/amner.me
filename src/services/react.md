---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.