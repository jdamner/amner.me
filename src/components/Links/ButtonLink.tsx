import Link from "next/link";
import { ButtonClasses } from "./Button";
import { useState } from "react";
import Spinner from "../Global/Spinner";
import { useRouter } from "next/router";
import { event } from "../../api/Insights";

export default function ButtonLink(props): JSX.Element {

	const [isClicked, setIsClicked] = useState(false);
	const router = useRouter();

	const handleClick = function (e: React.MouseEvent<HTMLAnchorElement>) {
		event('click', { name: e.currentTarget.innerText });
		setIsClicked(true);
		router.events.on('routeChangeComplete', () => {
			setIsClicked(false);
		});
	}

	return (
		<Link className={ButtonClasses.join(' ')} {...props} onClick={handleClick}>
			{ isClicked ? 
				<Spinner className='text-center' /> : props.children }
		</Link>
	)
}