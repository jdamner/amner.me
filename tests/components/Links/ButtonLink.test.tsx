import { render } from "@testing-library/react"
import ButtonLink from "../../../src/components/Links/ButtonLink"

describe("ButtonLink", () => {
  it("renders correctly", () => {
    const { container } = render(<ButtonLink href='/'>Test</ButtonLink>)
    expect(container).toMatchSnapshot()
  })
})
