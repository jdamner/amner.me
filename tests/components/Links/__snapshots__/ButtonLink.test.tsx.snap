// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders correctly 1`] = `
<div>
  <a
    class="border-solid border-2 border-slate-800 uppercase font-black font-sans bg-white px-4 py-2 text-slate-800 hover:bg-slate-100 hover:text-black"
    href="/"
  >
    Test
  </a>
</div>
`;
